internal/encryption/encryption.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2024 Canonical Ltd.
+// SPDX-License-Identifier: AGPL-3.0
+
+package encryption
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"encoding/hex"
+	"fmt"
+	"io"
+
+	"github.com/canonical/identity-platform-admin-ui/internal/logging"
+)
+
+type Encrypt struct {
+	gcm cipher.AEAD
+
+	logger logging.LoggerInterface
+}
+
+// Encrypt takes a plain string and returns a hex encoded string
+func (e *Encrypt) Encrypt(data string) (string, error) {
+	payload := []byte(data)
+	nonce, err := e.generateCipherNonce()
+	if err != nil {
+		e.logger.Errorf("error generating random the nonce, %v", err)
+		return "", fmt.Errorf("error generating random the nonce, %v", err)
```

Comment:
```suggestion
     err = fmt.Errorf("error generating random the nonce, %v", err)
		e.logger.Errorf(err.Error())
		return "", err
```

this might do too